File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/services/ProgressServiceKeeper.java
package net.kdt.pojavlaunch.services;

import android.content.Context;

import net.kdt.pojavlaunch.progresskeeper.TaskCountListener;

/**
 * タスク数を監視し、タスクが存在する場合にProgressServiceを開始するリスナー。
 */
public class ProgressServiceKeeper implements TaskCountListener {
    private final Context context;

    /**
     * @param ctx サービス開始に使用するコンテキスト
     */
    public ProgressServiceKeeper(Context ctx) {
        this.context = ctx;
    }

    /**
     * タスク数が0より大きい場合にProgressServiceを開始します。
     */
    @Override
    public void onUpdateTaskCount(int taskCount) {
        if(taskCount > 0) ProgressService.startService(context);
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/feature/ProgressServiceKeeper.kt
package com.arata.yukarilauncher.feature

import android.content.Context
import com.arata.yukarilauncher.task.TaskCountListener

/** タスク開始時にProgressServiceを起動するためのリスナークラス。 */
class ProgressServiceKeeper(private val context: Context) : TaskCountListener {

    /** タスクカウント更新時にサービスを開始する。 @param taskCount 現在のタスク数 */
    override fun onUpdateTaskCount(taskCount: Int) {
        if (taskCount > 0) ProgressService.startService(context)
    }
}
